system/jlib/jtrace.cpp

```diff
@@ -783,11 +756,33 @@ class CServerSpan : public CSpan
         setContextAttributes();
     }
 
-    void toLog(StringBuffer & out) const override
+    virtual const char* queryGlobalId() const override
+    {
+        return hpccGlobalId.get();
+    }
+
+    virtual const char* queryCallerId() const override
+    {
+        return hpccCallerId.get();
+    }
+
+    virtual const char* queryLocalId() const override
+    {
+        return hpccLocalId.get();
+    }
+
+    virtual void toLog(StringBuffer & out) const override
     {
         out.append("\"Type\":\"Server\"");
         CSpan::toLog(out);
 
+        if (!isEmptyString(hpccGlobalId.get()))
```

Comment:
Moving these to serverSpans makes sense. I will mention, all these LN specific IDs have a very limited expected lifespan. One major request I've heard is to provide these values during "transition" period (from LN IDs to OTel IDs). My concern is not reporting these values for non server spans could make the mapping of LN -> Otel IDs unnecessarily difficult. If I've misunderstood the net effect of moving these up, please ignore this comment.

Reply:
They should still be reported exactly as before.  The difference is that they're not stored in any of the other spans.

Reply:
It might be that the intent was that if you logged a client span it would log the global id, but it didn't before.  That can be implemented as a separate PR if required (probably a sensible idea).